src/Compilers/Core/Portable/SourceGeneration/SyntaxContextReceiverAdaptor.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Wraps an <see cref="ISyntaxReceiver"/> in an <see cref="ISyntaxContextReceiver"/>
+    /// </summary>
+    internal sealed class SyntaxContextReceiverAdaptor : ISyntaxContextReceiver
+    {
+        public SyntaxContextReceiverAdaptor(ISyntaxReceiver? receiver)
```

Comment:
Why allow a `null` `receiver`?

Reply:
As written this is accounting for the case where the user delegate returns `null`. Could move that logic up into the code which creates the adapter and avoid the `null` here.